Rajat issues 8 9

rajat-chandra opened the pull request:
Addressing issues 8 and 9 for plotting figures 2 and 3

sopeeweje commented:
Thanks @rajat-chandra! Couple things I'm now realizing:
https://github.com/sopeeweje/NLP-AI-Medicine/blob/ef8a6f92d4793bf7de6d2cc50a772a1b69623011/random_plotting.py#L109-L124

These are the colors for Figure 3 assuming that there are 12 categories ordered as clinical vs. technical as indicated. It would probably be better to have a spectrum of red colors and a spectrum of blue colors auto-generated based on how many clinical and technical categories there are in the final clusters. For example, if there are 8 unique clinical categories and 10 unique technical categories, there should be a spectrum of 8 red and a spectrum of 10 blues to use for the plot. To do this, we could add a 3rd column to final data ("Clinical/Technical") that denotes whether the category is clinical or technical then assign an appropriate color to each category (similar to how @sb1024 added Description and Category). 

To address your comment on #8:
1. Sounds good assuming that's how @sb1024 added them to that structure
2. Yeah that's always annoying, don't worry about that
3. I messed around with those annotations for hours for the original plot, but honestly I don't think we even need that figure anymore so don't worry about it
4. You can just rename it
5. No worries we can keep the color coding for now
6. Some of them will be the "N/A" category more than likely - for a final set of 60 clusters there will probably be 4-8 that are so weakly defined (subjectively and by silhouette score) that I termed them "nondescript" and excluded them from the analysis. So you can add these lines back in:
https://github.com/sopeeweje/NLP-AI-Medicine/blob/43b0a0fcda291f9fa0578afa31a676423b29234a/random_plotting.py#L112-L113

sb1024 commented:
Because there are a few additional columns from my previous edits (availability, etc.) these 2 new columns are at indices 18+19. I am adding the other new columns (mechanisms, for issue #5) at the end so there should be no influence (but of note if another new column is to be added)

rajat-chandra commented:
Thanks @sopeeweje and @sb1024 ! I have a few follow up points:
1. Updated description and category indices to 18 and 19
4. Renamed random_plotting.py to plotting.py
6. Indicated to continue if category is not ‘N/A’

I also updated analyze_clusters.py to add the “Clinical/Technical” column as the second last index (20), initialized with “ “ when writing to the csv, as was done for “Description” and “Category”. The “Clinical/Technical” column is assumed to have values of “Clinical” or “Technical”, and a warning is issued if this is not the case.

Is there anything else I should add here for the “Clinical/Technical” column and are there any conflicts with modifications you’re making @sb1024? I can adjust the index / make any changes to make it compatible if so!

Lastly, I worked on automatically creating the color gradients:
https://github.com/sopeeweje/NLP-AI-Medicine/blob/9200e5be363f9cdf0cf3dd28dd3e5c293dab25d7/plotting.py#L108-L137

https://github.com/sopeeweje/NLP-AI-Medicine/blob/9200e5be363f9cdf0cf3dd28dd3e5c293dab25d7/plotting.py#L170-L186

![image](https://user-images.githubusercontent.com/61924068/136140758-634bd6fc-eaa6-4c14-be88-e64f11040f1d.png)

Not sure why the legends aren’t formatted the same as in your plot since I didn’t change any of the plotting parameters, but if it looks off when you run it with the final data I can try to make adjustments. 

If everything looks ok I can merge it in. Thanks!
